code/modules/buildmode/bm_mode.dm

```diff
@@ -0,0 +1,86 @@
+/datum/buildmode_mode
+	var/key = "oops"
+
+	var/datum/click_intercept/buildmode/BM
+
+	var/use_corner_selection = FALSE
+	var/processing_selection = FALSE
+	var/list/preview
+	var/turf/cornerA
+	var/turf/cornerB
+
+/datum/buildmode_mode/New(datum/click_intercept/buildmode/newBM)
+	BM = newBM
+	preview = list()
+	return ..()
+
+/datum/buildmode_mode/Destroy()
+	Reset()
+	return ..()
+
+/datum/buildmode_mode/proc/enter_mode(datum/click_intercept/buildmode/BM)
+	return
+
+/datum/buildmode_mode/proc/exit_mode(datum/click_intercept/buildmode/BM)
+	return
+
+/datum/buildmode_mode/proc/get_button_iconstate()
+	return "buildmode_[key]"
+
+/datum/buildmode_mode/proc/show_help(mob/user)
+	CRASH("No help defined, yell at a coder")
```

Comment:
CRASH won't give the user a message - it'll just print a debug log. Is this intentional?

Reply:
Oh, I didn't consider that - I'll add a message to the user

Reply:
Corrected